### DISCUSSION TITLE
Lora Simple RX/TX WL55 problem

### DISCUSSION THREAD

**Discussion by alirezadh2** [micropython/micropython#13184] [STM32 / Pyboard]
I used two Nucleo_WL55 boards and python code for simple lora communication and it does not working ( time out):

Send:

from umachine import Pin, SPI
import time
import pyb
import lora
import micropython

micropython.alloc_emergency_exception_buf(256)

def get_sync_modem():
	from lora import WL55SubGhzModem
    	lora_cfg = { 'freq_khz': 868000 }
	return WL55SubGhzModem(lora_cfg=lora_cfg)    

def main_task():
	counter=0
	modem = get_sync_modem()
	modem.calibrate()
	while True:
       	        print("Sending...")
                modem.send(f"Hello from AEMICS #{counter}".encode(),tx_at_ms=None)
                print("Sent!")
	        pyb.LED(1).toggle()
               

if __name__ == "__main__":
	main_task() 

Receive:

from umachine import Pin, SPI
import time
import pyb
import lora
import micropython

micropython.alloc_emergency_exception_buf(256)

def get_sync_modem():

	from lora import WL55SubGhzModem
    	lora_cfg = { 'freq_khz': 868000 }
	return WL55SubGhzModem(lora_cfg)  

def main_task():

	modem = get_sync_modem()
	modem.calibrate()
	modem.rx_crc_error = True
	while True:
		print("Receiving...")
		rx = modem.recv(timeout_ms=10000, rx_length=0xFF, rx_packet=None)
       		if rx:
         		print(f"Received: {repr(rx)}")
       	        else:
			print("Receive timeout!")
	        pyb.LED(1).toggle()
               

if __name__ == "__main__":
	main_task() 

I got this errors:
Receiving...
_radio_isr
RX flags 0x200
Receive timeout!

**Comment by davefes** [comment]
Try to re-edit your post using code tags.   Start with 3 back-ticks py, line feed, your code, line feed and finish with 3 back ticks.  Have you had the Simple RX/TX example working?  Your sender method doesn't have any delays in it.

**Reply by davefes** [reply]
I am interested that you do a `modem.calibrate()` I did not read about that requirement.

Are you running the sender method on one unit and the receiver method on the 2nd NUCLEO?

**Reply by davefes** [reply]
Also, be aware of the limitation on file size upload on the NUCLEO_WL55.
https://github.com/orgs/micropython/discussions/13160
Small files <1KB will probably work.

**Reply by alirezadh2** [reply]
Thanks for reply, yeah I used one unit for sending and one for receiving, it doesn't receive any thing

**Comment by alirezadh2** [comment]
I remove modem.calibrate() and also add delay on sending still not working:
```py

#Send:

from machine import Pin, SPI
import time
import pyb
import lora
import micropython

micropython.alloc_emergency_exception_buf(256)

def get_sync_modem():
	from lora import WL55SubGhzModem
    	lora_cfg = { 'freq_khz': 868000 }
	return WL55SubGhzModem(lora_cfg=lora_cfg)    

def main_task():
	counter=0
	modem = get_sync_modem()
	while True:
       	        print("Sending...")
                modem.send(f"Hello from AEMICS #{counter}".encode(),tx_at_ms=None)
		time.sleep_ms(1000)
                print("Sent!")
	        pyb.LED(1).toggle()
               

if __name__ == "__main__":
	main_task() 



#Receive:


from machine import Pin, SPI
import time
import pyb
import lora
import micropython

micropython.alloc_emergency_exception_buf(256)

def get_sync_modem():

	from lora import WL55SubGhzModem
    	lora_cfg = { 'freq_khz': 868000 }
	return WL55SubGhzModem(lora_cfg)  

def main_task():

	modem = get_sync_modem()
	modem.rx_crc_error = True
	while True:
		print("Receiving...")
		rx = modem.recv(timeout_ms=5000, rx_length=0xFF, rx_packet=None)
       		if rx:
         		print(f"Received: {repr(rx)}")
       	        else:
			print("Receive timeout!")
	        pyb.LED(1).toggle()
               

if __name__ == "__main__":
	main_task() 

```

**Comment by davefes** [comment]
The Simple RX/TX example I was following uses:
```py
modem = get_modem()
```
On 2nd thought maybe it is smart enough to work out which modem you have installed.

**Reply by davefes** [reply]
Also, this might help:
https://github.com/orgs/micropython/discussions/13180
especially doing `mpremote mip install lora-sync` ... it was just updated.

**Reply by alirezadh2** [reply]
I am using sync modem, your setup is working with Async modem?

**Reply by alirezadh2** [reply]
your setup is Nucelo_WL55?

**Reply by davefes** [reply]
I have only just started this "journey" ... still using sync modem.  Some day I may have to do a Async project.

Most of my current work on LoRa is using the WeACT BlackPill using this repro:
https://github.com/nspsck/STM32F411CEU6_BlackPill_Micropython/tree/main

I did spend a few days trying to get micropython/lora to work on a E77-400M22S, which uses a STM32WL5x part.  I could not debug my application as that part and evidently the NUCLEO_WL55 will not allow you to upload files bigger than about 1KB.

**Reply by alirezadh2** [reply]
I am also trying to figure out more details in micropython files about lora libraries.
